webui/react/src/globalStorage.ts

```diff
@@ -0,0 +1,42 @@
+import { Storage } from 'utils/storage';
+
+class GlobalStorage {
+  private storage: Storage;
+  private keys: Record<string, string>;
+
+  constructor(storage: Storage) {
+    this.storage = storage;
+    this.keys = {
+      authToken: 'auth-token',
+      serverAddress: 'server-address',
+    };
+  }
+
+  set authToken(token: string) {
+    this.storage.set(this.keys.authToken, token);
+  }
+
+  getAuthToken() {
```

Comment:
**suggestion:** any reason not to follow the getter pattern? similar to the setter pattern you have for `set authToken`?
```
get authToken() {
  return this.storage.get<string>(this.keys.authToken);
}
```

Reply:
I was running into a return type issue here. I'll take another look

Reply:
it worksout fine